apps/explorer/src/app/components/txs/txs-infinite-list.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { FixedSizeList as List } from 'react-window';
+import InfiniteLoader from 'react-window-infinite-loader';
+import { t } from '@vegaprotocol/react-helpers';
+import { TxsInfiniteListItem } from './txs-infinite-list-item';
+import type { ChainExplorerTxResponse } from '../../routes/types/chain-explorer-response';
+
+interface TxsInfiniteListProps {
+  hasMoreTxs: boolean;
+  areTxsLoading: boolean | undefined;
+  txs: ChainExplorerTxResponse[] | undefined;
+  loadMoreTxs: () => void;
+  error: Error | undefined;
+  className?: string;
+}
+
+interface ItemProps {
+  index: number;
+  style: React.CSSProperties;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NOOP = () => {};
+
+export const TxsInfiniteList = ({
+  hasMoreTxs,
+  areTxsLoading,
+  txs,
+  loadMoreTxs,
+  error,
+  className,
+}: TxsInfiniteListProps) => {
+  if (!txs) {
+    return <div>No items</div>;
+  }
+
+  // If there are more items to be loaded then add an extra row to hold a loading indicator.
+  const itemCount = hasMoreTxs ? txs.length + 1 : txs.length;
+
+  // Pass an empty callback to InfiniteLoader in case it asks us to load more than once.
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const loadMoreItems = areTxsLoading ? NOOP : loadMoreTxs;
+
+  // Every row is loaded except for our loading indicator row.
+  const isItemLoaded = (index: number) => !hasMoreTxs || index < txs.length;
+
+  const Item = ({ index, style }: ItemProps) => {
```

Comment:
question (non-blocking): seems like a very strange pattern for me to define a component inside a component, which will then has to redefine itself every time the list re-renders. 

my intuition would way to put this outside of the component, and do something like:

```

const Item = ({ index, style, isLoading, error }: ItemProps) => (
 ...
)

const TxsInfiniteList = (...) => {
  return (
    ...
      <List {...}>
        {({ index, style }) => <Item index={index} style={style} isLoading={!isItemLoaded(index)} />}
      </List>
    ...
   )
 }
```

Reply:
Tbh I copied the boilerplate from react-window-infinite-loader's example. I agree your suggestion is better, thanks.